pytorch_lightning/utilities/device_dtype_mixin.py

```diff
@@ -48,9 +48,11 @@ def device(self) -> Union[str, torch.device]:
         return device
 
     @device.setter
-    def device(self, new_device: Union[str, torch.device]):
-        # Necessary to avoid infinite recursion
-        raise RuntimeError('Cannot set the device explicitly. Please use module.to(new_device).')
+    def device(self, device: Union[str, torch.device]):
+        if device.type == "xla":
+            raise MisconfigurationException("xla should be set using device")
+        self._device = device
+        self.to(device)
```

Comment:
not sure why this setter exists. can we delete it? 
I can't see it being used anywhere.

Reply:
Done !